packages/vuetify/src/components/VDataIterator/VDataIterator.ts

```diff
@@ -67,26 +67,28 @@ export default mixins(Themeable).extend({
   },
 
   watch: {
-    value (value: any[]) {
-      this.selection = value.reduce((selection, item) => {
-        selection[getObjectValueByPath(item, this.itemKey)] = true
-        return selection
-      }, {})
-    },
-    selection: {
-      handler (value: Record<string, boolean>, old: Record<string, boolean>) {
-        if (deepEqual(value, old)) return
-        const keys = Object.keys(value).filter(k => value[k])
-        const selected = !keys.length ? [] : this.items.filter(i => keys.includes(String(getObjectValueByPath(i, this.itemKey))))
-        this.$emit('input', selected)
+    value: {
+      handler (value: any[]) {
+        this.selection = value.reduce((selection, item) => {
+          selection[getObjectValueByPath(item, this.itemKey)] = item
+          return selection
```

Comment:
@nekosaur Throwing `[Vue warn]: Invalid prop: type check failed for prop "value". Expected Boolean, got Object ` error